dtfs-central-api/api-tests/v1/utilisation-reports/post-upload-utilisation-report.api-test.ts

```diff
@@ -127,13 +127,37 @@ describe(`POST ${getUrl()}`, () => {
     expect(azureFileInfo).toHaveLength(1);
   });
 
-  it('saves the report when there is a large number of fee records', async () => {
+  it('saves the report when there is a large number of fee records all for the same facility', async () => {
     // Arrange
     const numberOfReportDataEntriesToCreate = 500;
 
+    const facilityId = '12345678';
+    const reportData: UtilisationReportRawCsvData[] = [];
+    while (reportData.length < numberOfReportDataEntriesToCreate) {
+      reportData.push({ ...aUtilisationReportRawCsvData(), 'ukef facility id': facilityId });
+    }
+
+    const payload: PostUploadUtilisationReportRequestBody = { ...aValidPayload(), reportData };
+
+    // Act
+    const response = await testApi.post(payload).to(getUrl());
+
+    // Assert
+    expect(response.status).toBe(HttpStatusCode.Created);
+
+    const numberOfInsertedFeeRecords = await SqlDbHelper.manager.count(FeeRecordEntity, {});
```

Comment:
can this be in a different it statement please

Reply:
I'd rather keep this all in the same test, but this assertion doesn't actually need to be in this test so I have removed it

Reply:
EDIT: I have decided to keep these assertions in as they come under the statement of "saves the report". Please let me know if you disagree

Reply:
I think it should be different it statements.  If one fails, it is easier to see exactly which is failing from the it statement description

Reply:
Jest allows you to see exactly which part of the test failed. In the case where you don't get a `201`, that will be very clear from the test output. Similarly, in the case where you do get a `201`, it would be clear that it was the number of fee records inserted which caused this to fail.

I actually think having different `it` blocks would make things less clear. For instance, the it block which says "saves the report" may pass if a `201` is returned, but the below test block which asserts that the correct number of fee records is saved may fail. This is not semantically accurate as the return of a `201` accompanied by saving an incorrect number of fee records implies that "saving the report" did not actually work successfully, but the separate `it` blocks may cause the debugger to believe that "saving the report" was successful and so the error must be elsewhere

Reply:
This still should be split.  It allows for future proofing as well incase we wanted to add more tests for this bit.  we can change the it statement to make more sense - eg it('returns a status of ${HttpStatusCode.Created}' ... ) and it('should have correct number of records...).  Can have a describe around it too.